rmf_site_format/src/camera_poses.rs

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+use crate::{Anchor, Category, Pose, Rotation};
+#[cfg(feature = "bevy")]
+use bevy::prelude::Component;
+use glam::{Affine3A, Quat, Vec3};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "bevy", derive(Component))]
+pub struct CameraPoses(pub HashMap<String, Pose>);
```

Comment:
I think it would be more idiomatic if we had something like

```
#[derive(Bundle)]
pub struct UserCameraPose {
    pose: Pose,
    name: NameInSite,
    marker: UserCameraPoseMarker,
}

#[derive(Component)]
pub struct UserCameraPoseMarker;
```

and then each UserCameraPose entity is a child of either a level or the site depending on whether it's meant to apply to a single level or to any level within the site.

Or if we want to give users a high degree of configuration for these, we could imitate the [`AssociatedGraphs`](https://github.com/open-rmf/rmf_site/blob/269efabaa7def79aa3d827606e2cd21d67960dc3/rmf_site_format/src/nav_graph.rs#L67-L71) component used by nav graphs, e.g.

```
#[derive(Component)]
pub struct AssociatedLevels<T> {
    All,
    Only(BTreeSet<T>),
    AllExcept(BTreeSet<T>),
}
```

so then we just have one group of UserCameraPoses defined at the site level, and each UserCameraPose specifies which levels it's meant to apply to.

Reply:
As a separate concern, I'm worried that the name `CameraPoses` will be too ambiguous: Is it referring to simulated camera device poses, or the pose of the application view seen by the user? I think we should add `User` into the name to make the purpose more explicit.

Reply:
I went for the simplest (Bundle) way for now. A larger refactor can be done as part of the effort to address the feature request. https://github.com/open-rmf/rmf_site/pull/215/commits/abb754406b78d8720c78f11d86950227496528cc
For now I only added the storage for the `UserCameraPose` as a `BTreeMap<u32, UserCameraPose>` in levels, it should be trivial to extend that to sites as well once site wide poses are available and a UI to edit them is ready.